controller/controller.go

```diff
@@ -116,7 +116,12 @@ func (ctrl *baseController) init(ctx context.Context) error {
 func InstallDriver(ctx context.Context, conf *config.Config) error {
 	if conf.Driver.Terraform != nil {
 		tfConf := *conf.Driver.Terraform
-		return driver.InstallTerraform(ctx, *tfConf.Path, *tfConf.WorkingDir)
+		err := driver.InstallTerraform(ctx, *tfConf.Path, *tfConf.WorkingDir)
```

Comment:
Maybe check the Terraform version before downloading/installing it?

Reply:
the UX of this is a bit awkward tbh. Users can't set which Terraform version to run yet (just the latest 0.13), and the work-around is to manually install the desired version and point to it. An existing Terraform at the path would skip the install in fetch its version to then compare with the config.

An improved order would be something like:
* Configured version > validate compat > install
* No configured version > check existing version > validate compat > install
* No configured version > no existing version > fetch latest version > validate compat > install

I had made these changes with minimal impact in mind but you pointing this out makes it more convincing to reorganize the code so it's more logical.

Reply:
I thought there might be a reasons but digging around I couldn't find them. I'm cool with it as a working solution if you want to move forward for now and circle back to this later.

I'll approve this PR and leave that to you.

Reply:
Appreciate it John! I'll merge it and iterate on this as a follow-up.